### PR TITLE
Cleanup many docstrings.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

A bit long, but mostly on the mechanical side of things.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tri/triangulation.py [hunk 1]
```diff
@@ -193,8 +193,11 @@ def neighbors(self):
 
     def set_mask(self, mask):
         """
-        Set or clear the mask array.  This is either None, or a boolean
-        array of shape (ntri).
+        Set or clear the mask array.
+
+        Parameters
+        ----------
+        mask : None or bool array or shape (ntri,)
```
Comment (timhoffm) [inline lib/matplotlib/tri/triangulation.py]:
```suggestion
        mask : None or bool array of length (ntri)
```

File: lib/matplotlib/tri/trirefine.py [hunk 2]
```diff
@@ -77,12 +77,11 @@ def refine_triangulation(self, return_tri_index=False, subdiv=3):
...
-        return_tri_index : boolean, optional
-            Boolean indicating whether an index table indicating the father
-            triangle index of each point will be returned. Default value
-            False.
-        subdiv : integer, optional
-            Recursion level for the subdivision. Defaults value 3.
+        return_tri_index : bool, optional, default: False
+            Whether an index table indicating the father triangle index of each
+            point will be returned.
+        subdiv : integer, optional, default: 3
```
Comment (timhoffm) [inline lib/matplotlib/tri/trirefine.py]:
```suggestion
        subdiv : int, optional, default: 3
```

File: lib/matplotlib/widgets.py [hunk 3]
```diff
@@ -1997,7 +1997,7 @@ def __init__(self, ax, onselect, drawtype='box',
         and *minspanx* will be interpreted in the same coordinates as
         the x and y axis. If 'pixels', they are in pixels.
 
-        *button* is a list of integers indicating which mouse buttons should
+        *button* is the list of (integer) mouse buttons which should
```
Comment (timhoffm) [inline lib/matplotlib/widgets.py]:
```suggestion
        *button* is the list of `.MouseButton`\s which should
```
Will additionally need to make it a raw string for the escape backslash.

File: lib/matplotlib/backend_bases.py [hunk 4]
```diff
@@ -1061,13 +1059,11 @@ class TimerBase:
...
     Attributes
     ----------
-    interval : scalar
-        The time between timer events in milliseconds. Default is 1000 ms.
-
-    single_shot : bool
-        Boolean flag indicating whether this timer should operate as single
-        shot (run once and then stop). Defaults to `False`.
-
+    interval : scalar, default: 1000ms
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/backend_bases.py]: Sure, all changed to float/int (as appropriate).
I think the unitful version is much easier to read unless you're a robot parsing the docstring -- but in that case you may as well just look at the actual function signature...

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -4361,15 +4347,15 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
 
             - 'face': The edge color will always be the same as the face color.
             - 'none': No patch boundary will be drawn.
-            - A Matplotlib color or sequence of color.
+            - A color or sequence of colors.
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: It can be a single color?

**Comment by anntzer** [comment]
thanks, all fixed

**Comment by timhoffm** [comment]
I'm not trying to backport because this will for sure raise merge conflicts.

**Pull request merged**